app/src/main/java/com/puzzlebench/cleanMarvelKotlin/presentation/mvp/CharacterPresenter.kt

```diff
@@ -21,87 +21,84 @@ const val COMIC_URL_TAG = "comiclink"
 open class CharacterPresenter(view: CharacterView,
                               private val getCharacterServiceUseCase: GetCharacterServiceUseCase,
                               private val getCharacterDetailServiceUseCase: GetCharacterDetailServiceUseCase,
-                              private val getFromDatabaseUseCase: GetFromDatabaseUseCase,
-                              private val addToDatabaseUseCase: AddToDatabaseUseCase,
+                              private val getCharacterDatabaseUseCase: GetCharacterDatabaseUseCase,
                               private val subscriptions: CompositeDisposable) : Presenter<CharacterView>(view) {
 
     fun init() {
         view.init()
-        requestGetCharacters()
-        getCharacter()
-        refresh()
+        retrieveFromDatabase()
+        listenToRefresh()
+        listenToDetails()
     }
 
-    private fun refresh() {
+    private fun listenToRefresh() {
         RxBus.subscribe(view.activity, object : OnRefreshPressedBusObserver() {
```

Comment:
now that you know RXAndroid you can move away from RxBus

Reply:
done!